warehouse/query-core/src/main/java/datawave/query/planner/DefaultQueryPlanner.java

```diff
@@ -469,9 +477,21 @@ protected CloseableIterable<QueryData> process(ScannerFactory scannerFactory, Me
                 cfg = getQueryIterator(metadataHelper, config, settings, "", false, false);
             }
             configureIterator(config, cfg, newQueryString, isFullTable);
+
+            while (null == logCfg) {
+                logCfg = getQueryLogIterator(config, settings);
+                if (logSettingFuture.isDone()) {
+                    // It's possible that getQueryLogIterator will return a non-null setting once we know the future is done. Check one more time.
+                    if (logCfg == null) {
+                        logCfg = getQueryLogIterator(config, settings);
+                    }
+                    break;
+                }
+            }
         }
 
-        final QueryData queryData = new QueryData().withQuery(newQueryString).withSettings(Lists.newArrayList(cfg));
+        List<IteratorSetting> iteratorSettings = logCfg == null ? Collections.singletonList(cfg) : Lists.newArrayList(cfg, logCfg);
```

Comment:
We should only be adding the logCfg if config.isTserverLoggingActive().

Reply:
Optionally avoid creating the logCfg altogether if config.IsTserverLoggingActive() == false;.,